File: RentAppProject/RentApp.Shared/Models/QualityReportItem.cs
using System;
namespace RentAppProject
{
	public class QualityReportItem
	{
        public int ID { get; set; }
		public int CleaningTaskID { get; set; }
		public int QualityReportID { get; set; }
		public int Rating { get; set; }
		public string Comment { get; set; }
        public string ImageLocation { get; set; }
        public CleaningTask CleaningTask { get; set; }
		
        public QualityReportItem() { }
		public QualityReportItem(CleaningTask area, int reportID)
		{
			this.CleaningTask = area;
			this.CleaningTaskID = area.ID;
			this.QualityReportID = reportID;
		}
	}
}

File: Itido-BMS/dart/bms_dart/lib/src/models/quality_report_item.dart
import 'task.dart';

class QualityReportItem {
  int id;
  int rating;
  String comment;
  String imageLocation;
  Task task;

  QualityReportItem({
    this.id,
    this.rating,
    this.comment,
    this.imageLocation,
  });

  factory QualityReportItem.fromJson(json) {
    if (json == null) return null;
    return QualityReportItem._fromJson(json);
  }

  QualityReportItem._fromJson(json)
      : this.id = json['id'],
        this.rating = json['rating'],
        this.comment = json['comment'],
        this.imageLocation = json['imageLocation'],
        this.task = Task.fromJson(json['task']);

  Map<String, dynamic> toMap() => {
        'id': this.id,
        'rating': this.rating,
        'comment': this.comment,
        'imageLocation': this.imageLocation,
      };
}
